Apps/CommonData/src/Utils/AddressUtility.cs

```diff
@@ -15,15 +15,267 @@
 // -------------------------------------------------------------------------
 namespace HealthGateway.Common.Data.Utils
 {
+    using System;
     using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
     using System.Linq;
+    using System.Text.RegularExpressions;
+    using HealthGateway.Common.Data.Constants;
     using HealthGateway.Common.Data.Models;
 
     /// <summary>
-    /// Utilities for extracting and constructing values to and from a client registry address.
+    /// Utilities for interacting with addresses.
     /// </summary>
-    public static class AddressUtility
+    public static partial class AddressUtility
     {
+        /// <summary>
+        /// Regular expression for Canadian postal codes.
+        /// </summary>
+        [GeneratedRegex(@"^[A-Z]\d[A-Z] \d[A-Z]\d$")]
+        public static partial Regex PostalCodeRegex();
+
+        /// <summary>
+        /// Regular expression for American ZIP codes and ZIP+4 codes.
+        /// </summary>
+        [GeneratedRegex(@"^\d{5}(-\d{4})?$")]
+        public static partial Regex ZipCodeRegex();
+
+        /// <summary>
+        /// Canadian province and territory abbreviations ordered alphabetically by province name.
+        /// </summary>
+        public static readonly IEnumerable<ProvinceAbbreviation> ProvinceAbbreviations = new List<ProvinceAbbreviation>
+        {
+            ProvinceAbbreviation.AB, ProvinceAbbreviation.BC, ProvinceAbbreviation.MB, ProvinceAbbreviation.NB, ProvinceAbbreviation.NL, ProvinceAbbreviation.NT, ProvinceAbbreviation.NS,
+            ProvinceAbbreviation.NU, ProvinceAbbreviation.ON, ProvinceAbbreviation.PE, ProvinceAbbreviation.QC, ProvinceAbbreviation.SK, ProvinceAbbreviation.YT,
+        };
+
+        /// <summary>
+        /// American state abbreviations ordered alphabetically by state name.
+        /// </summary>
+        public static readonly IEnumerable<StateAbbreviation> StateAbbreviations = new List<StateAbbreviation>
```

Comment:
No need to change this as they're pretty constant, but I am curious why you didn't do this programatically?

https://stackoverflow.com/questions/46718622/how-to-order-enum-alphabetically-rather-than-on-int

The same could be said about the next value where order all countries without CA and US and then splice them to the front.

Reply:
I might have just forgot about `Enum.GetValues()`.

Reply:
Resolved now.  Much tidier.